bukkit/src/main/java/com/github/games647/fastlogin/bukkit/listener/ConnectionListener.java

```diff
@@ -45,6 +50,57 @@ public void onPlayerJoin(PlayerJoinEvent joinEvent) {
             // cases: Paper (firing BungeeCord message before PlayerJoinEvent) or not running BungeeCord and already
             // having the login session from the login process
             BukkitLoginSession session = plugin.getSession(player.getAddress());
+            
+            if(Bukkit.getServer().getPluginManager().getPlugin("Geyser-Spigot") != null &&
+                    Bukkit.getServer().getPluginManager().getPlugin("floodgate-bukkit") != null) {                
+                FloodgatePlayer floodgatePlayer = null;
+                
+                // check if the player is really connected through Geyser
+                for (GeyserSession geyserPlayer : GeyserConnector.getInstance().getPlayers()) {
+                    if (geyserPlayer.getName().equals(player.getName())) {
+                        // this also returns a floodgatePlayer for linked Java accounts
+                        // that's why the Geyser Server's player list also has to be checked
```

Comment:
According to my tests, `FloodgateAPI.getPlayer(player.getUniqueId())` also returns a `FloodgatePlayer` instance if the player is **connecting from Java Edition** (JE) but has a linked Bedrock Edition (BE) account.
Later in the code `floodgatePlayer != null` is used to decide if a player is connecting from BE. But since JE accounts with linked BE accounts also have a return value, I had to filter them out. That's what the for loop is used for.
I couldn't find a better solution (yet), but honestly, I didn't spend much time looking. I am planning to figure out an alternative method for this. If I won't find anything better, than at least I'll try switching `name` to `uuid` in [line 60](https://github.com/games647/FastLogin/pull/494/files/e1af75191edf9bce5e9d9b441b85b36f535eee0f#diff-554627ddde4f64e3c694783b936aef722dc908f261e03813733df48483e39f49R60).